tuna/helper.py

```diff
@@ -179,6 +180,24 @@ def mysqldb_overwrite_table(table, dict_list, filter_cols):
   return ret, insert_ids
 
 
+def session_retry(session, callback, actuator, logger=LOGGER):
+  """insert/update dict obj to mysql table"""
+  for idx in range(NUM_SQL_RETRIES):
+    try:
+      return actuator(callback)
+    except OperationalError as error:
+      logger.warning('%s, maybe DB contention sleeping (%s)...', error, idx)
+      session.rollback()
+      sleep(random.randint(1, 30))
+    except pymysql.err.OperationalError as error:
+      logger.warning('%s, maybe DB contention sleeping (%s)...', error, idx)
+      session.rollback()
+      sleep(random.randint(1, 30))
+
+  logger.error('All retries have failed.')
+  return None
+
+
```

Comment:
there is a db_helper.py in utils, maybe this would be better in there?

Reply:
There are some other functions in helper with similar characteristics. Perhaps these should be migrated as well.